internal/ui/keyboard_js.go

```diff
@@ -0,0 +1,39 @@
+// +build js
+
+package ui
+
+import "github.com/gopherjs/gopherjs/js"
+
+var doc = js.Global.Get("document")
+
+func push(char rune) {
+	go func() {
+		rblock.Lock()
+		runebuffer = append(runebuffer, char)
+		rblock.Unlock()
+	}()
+}
+
+func init() {
+	if doc == nil {
+		return
+	}
+	kp := doc.Get("onkeypress")
+	switch kp.Length() {
+	case 0:
+		doc.Set("onkeypress", push)
```

Comment:
Yes, actually had to refactor because preventdefault in keydown prevents keypress from firing, keypress might need preventdefault but I think keyup's preventdefault is doing the right thing.